File: backend/dataset/levelOrder/levelorder_17.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        
        if(root==None):
            return []
           
        minorList = []
        ansList = []
        i = 1
        level = 0
        queue = [(root, level)]
        
        while(queue):
            node, level = queue.pop()
            if(level == i):
                ansList.append(minorList)
                minorList = []
                i += 1
            minorList.append(node.val)
            if(node.left):        
                queue.insert(0, (node.left, level+1))
            if(node.right):
                queue.insert(0, (node.right, level+1))
                
        if(minorList):
            ansList.append(minorList)
            
        return ansList

File: backend/CDataset/levelOrder/levelOrder_17.cpp
class Solution {
public:
    #include<vector>

    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == NULL) return {};
        vector<vector<int>>res;
        queue<TreeNode*>q;
        q.push(root);
        vector<int>temparr;
        while(!q.empty()) {
            // n is the number of nodes in the current level
            int n = q.size();
            for (int i = 0; i<n; i++) {
                TreeNode* temp = q.front();
                q.pop();
                temparr.push_back(temp->val);
                if (temp->left) {
                    q.push(temp->left);
                }
                if (temp->right) {
                    q.push(temp->right);
                }
            }
            res.push_back(temparr);
            temparr.clear();
        }
        return res;
    }
};